lib/perl/Genome/Model/Tools/Analysis/Coverage/MergeReadcounts.pm

```diff
@@ -210,17 +213,16 @@ sub execute {
     #create temp directory for munging
     my $tempdir = Genome::Sys->create_temp_directory();
     unless($tempdir) {
-        $self->error_message("Unable to create temporary file $!");
-        die;
+        die("Unable to create temporary file $!");
     }
     
     
     # reads the variant files and combine
     # for a hash with the keys chr, start, stop, ref, var, type
     my (%hash, @header);
-    for (my $n=0; $n<@variants; $n ++)
+    for (my $n=0; $n<@variant_files; $n ++)
     {
-	my $variant = $variants[$n];
+	my $variant = $variant_files[$n];
```

Comment:
There's nothing wrong with this at all but I wanted to share an alternative way to loop through `@variant_files` and `@sources` using [List::MoreUtils::each_array][1] in case you find it more expressive.

```
use List::MoreUtils qw(each_array);

my $ea = each_array(@variant_files, @sources);
while (my ($variant, $source) = $ea->()) {
     ...
}
```

[1]: https://metacpan.org/pod/List::MoreUtils#each_array-ARRAY1-ARRAY2